src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -35,54 +37,59 @@ public Question(Long id, NsUser writer, String title, String contents) {
         this.writer = writer;
         this.title = title;
         this.contents = contents;
+        this.answers = new Answers();
     }
 
     public Long getId() {
         return id;
     }
 
-    public String getTitle() {
-        return title;
+    public NsUser getWriter() {
+        return writer;
     }
 
-    public Question setTitle(String title) {
-        this.title = title;
-        return this;
+    public void addAnswer(Answer answer) {
+        answer.toQuestion(this);
+        this.answers.add(answer);
     }
 
-    public String getContents() {
-        return contents;
+    public boolean isDeleted() {
+        return deleted;
     }
 
-    public Question setContents(String contents) {
-        this.contents = contents;
-        return this;
-    }
+    public List<DeleteHistory> delete(NsUser loginUser) throws CannotDeleteException {
+        try {
+            validateDelete(loginUser);
+            LocalDateTime deleteTime = LocalDateTime.now();
 
-    public NsUser getWriter() {
-        return writer;
-    }
+            List<DeleteHistory> deleteHistories = new ArrayList<>();
+            deleteHistories.add(delete(deleteTime));
+            deleteHistories.addAll(deleteAnswers(loginUser, deleteTime));
 
-    public void addAnswer(Answer answer) {
-        answer.toQuestion(this);
-        answers.add(answer);
+            return deleteHistories;
+        } catch (UnAuthorizedException e) {
+            throw new CannotDeleteException(e.getMessage());
+        }
     }
 
-    public boolean isOwner(NsUser loginUser) {
-        return writer.equals(loginUser);
-    }
+    private List<DeleteHistory> deleteAnswers(NsUser loginUser, LocalDateTime deleteTime) throws CannotDeleteException {
+        try {
+            return this.answers.deleteAnswers(loginUser, deleteTime);
+        } catch (UnAuthorizedException e) {
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+        }
 
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
     }
 
-    public boolean isDeleted() {
-        return deleted;
+    private DeleteHistory delete(LocalDateTime deleteTime) {
+        this.deleted = true;
+        return DeleteHistory.createQuestionHistory(this.id, this.writer, deleteTime);
     }
 
-    public List<Answer> getAnswers() {
-        return answers;
+    private void validateDelete(NsUser loginUser) {
+        if (!this.writer.equals(loginUser)) {
```

Comment:
```suggestion
        if (!this.writer.same(loginUser))) {
```

`NsUser`객체의 `equals` 함수를 재정의 하거나, 같냐는 메소드를 도출 해낼것 같아요.

Reply:
그러네요 객체에 메세지를 전달하는게 맞는데 자체에서 비교했군요..